src/simularium/WebsocketClient.ts

```diff
@@ -0,0 +1,317 @@
+import jsLogger from "js-logger";
+import { ILogger } from "js-logger";
+import { FrontEndError, ErrorLevel } from "./FrontEndError";
+
+export interface NetMessage {
+    connId: string;
+    msgType: number;
+    fileName: string;
+}
+// TODO: proposed new NetMessage data type:
+// This factors the raw data structure away from the networking and transmission info.
+// This allows the data structure to make a bit more sense with respect to typescript typing,
+// and also for raw file drag n drop it doesn't need connection info or msgtype.
+// interface NetMessage {
+//     connId: string; // unique connection to server
+//     msgType: number; // identifies the data structure of the message
+//     fileName: string; // identifies the trajectory this connection is dealing with
+//     payload: Object; // the JS object with the message data itself
+// }
+
+export interface MessageEventLike {
+    data: string;
+}
+
+// these have been set to correspond to backend values
+export const enum NetMessageEnum {
```

Comment:
Half baked idea: 
I wonder if the message ids should ultimately live inside the RemoteSimulator etc, and just keep this class for managing the connection state.  The json and binary message handlers would be much more generic then.  And anyone who wants to add a handler can have their own message interpreter.  You could still have this idea where each handler can either say "pass this message along to other handlers" or "stop, we have completely handled it" by returning a boolean or some other value type.

Reply:
at the same time, there is something very nice and simple about just adding handlers for each known message type right here...

Reply:
My only concern with moving these is making sure the values stay the same, since it's important for them to match with what octopus has. Like is it confusing if we define separate enums in separate files that are connecting to the WebsocketClient, ie in RemoteConverter we have something like
```
const enum ConverterEnum {
    ID_CONVERT_TRAJECTORY_FILE = 16,
    ID_CONVERTED_TRAJECTORY = 17,
}
```
Like is it weird to have this on its own separate from enums numbered 1-15 and 18+? Or am I overthinking this...